### ISSUE TITLE
--parallel doesn't respect verbosity

### ISSUE THREAD

**Issue by alanmcruickshank** [sqlfluff/sqlfluff#1065]
I've been doing some testing of the `--parallel` code and looking to see what's happening under the hood. At the moment the verbosity level (i.e. `-vvvvv`) isn't being respected within the threads. I think it *should* in some way, although I realise precisely *how* is a slightly trickier issue.

I think this is because the logging functions are controlled in `cli.py`, specifically `set_logging_level()`. Given the split into different processes happens in the linter, this isn't being re-called in each process. [caveat: this is a guess, not something I've confirmed].

I think the cli should control logging (because people importing sqlfluff as a module should have control of it themselves), however I also can see the value of multi processing being handles by the `linter`. I'd love a way for the cli to pass through logging configuration into the other processes such that they can report back appropriately.

@barrywhart - is this possible? (or even a good idea?)

The reason for seeing the logging would be for performance evaluation, so an alternative option here would be to refactor performance stats, so we get better feedback from each process rather than fixing the logging directly.

**Comment by barrywhart** [comment]
This makes a lot of sense, _especially_ in Windows.

As I mentioned in another PR, `multiprocessing` in Windows works **completely differently** from Unix-based OSes like Linux and MacOS. The latter OSes support `fork()`, where child processes begin life as a clone of the parent process, so they inherit global state such as logging. Windows does not, so child processes begin life as completely new processes, inheriting **none** of the transient state from the parent process. So we'll have to note any global state and pass it down or reinitialize it **explicitly**.

Is there any other global state we need to be thinking about as well?

My first thought is, perhaps we should create some sort of "state" or "config" object and pass that as a parameter. The child processes will use that to establish the same state as the parent. See #1060 (which addresses some other issues with `--parallel`) for a hint at how to do this. See the `_init_dialect()` function, which is called with `child_process=True` at the beginning of each child process. We could pass additional parameters to this function, potentially creating another function with a different name, since we're starting to diverge from dialect-specific stuff.

@alanmcruickshank: I would like to get #1060 reviewed and merged first before we work on this. The amount of code involved in parallel support is small, and having multiple PRs open at once will get messy because conflicts are likely. Do you have any interest in taking this issue? I'm happy to do it or for for you to look at it.

**Comment by alanmcruickshank** [comment]
Happy to merge #1060 first and then handle this as a separate PR. Especially given it's a potential windows issue, then it will be easiest for me to test it.

**Comment by barrywhart** [comment]
@alanmcruickshank: Great! Thanks for the review on #1060. I merged it just now. If you are reworking config in relation to parallel linting, you _might_ consider addressing #1058 at the same time.

**Comment by alanmcruickshank** [comment]
I think given my foray into parallelism, I'm going to close this issue as not necessary. I think what we've got is good enough.
